postgres-backup-template.yaml

```
@@ -61,7 +61,7 @@ objects:
                       PGPASSWORD="$DATABASE_PASSWORD" pg_dump --username=$DATABASE_USER --host=$DATABASE_HOST --port=$DATABASE_PORT --column-inserts --clean --create ${DATABASE_NAME} | gzip > /database-backup/$FILENAME); 
                       echo "";
                       echo "Backup successful"; du -h /database-backup/$FILENAME; 
-                      echo "to restore the backup to the serviced host use: $ psql --username=$DATABASE_USER --password --host=$DATABASE_HOST --port=$DATABASE_PORT postgres < /database-backup/<backupfile> (unpacked)"
+                      echo "to restore the backup to the serviced host use: $ zcat /database-backup/$MY-DUMP.tgz | PGPASSWORD=$DATABASE_PASSWORD psql --username=$DATABASE_USER --host=$DATABASE_HOST --port=$DATABASE_PORT postgres"
```

```suggestion
                      echo "to restore the backup to the serviced host use: $ zcat /database-backup/$MY-DUMP.tgz | PGPASSWORD=<database password> psql --username=$DATABASE_USER --host=$DATABASE_HOST --port=$DATABASE_PORT postgres"
```

Leave the password out of it, then imo this is ok to be merged. The reason is simple: This way, the password ends up in the container's logs. If it ends up in the container's logs, it might get sent to e.g. Elasticsearch or some other log storage. This is unexpected behaviour and might lead to the compromise of the password. Also, if someone went to great lengths and set up permissions so users cannot get the password in OpenShift, exposing it in the logs might compromise this again.

---

hmm ... the password is fetched by the env variable $DATABASE_PASSWORD .... so this should not appear in any logs, should it?

---

The environment variable is replaced by its value when writing to stdout. And what's written on stdout gets logged.